File: RepoTransBench/source_projects/Python/oxsecurity_MaskerLogger/public_tests/test_public_secrets_in_logs_example.py
import unittest
import logging
from maskerlogger.masker_formatter import MaskerFormatter

def log_sensitive_public():
    logger = logging.getLogger("test_logger_public")
    handler = logging.StreamHandler()
    handler.setFormatter(MaskerFormatter("%(message)s", regex_config_path=None))
    logger.addHandler(handler)
    logger.setLevel(logging.WARNING)
    logger.warning("secret field: thisshouldberedacted321 must be hidden")
    logger.warning("nondescript info log")
    logger.warning("skip me", extra={"apply_mask": False})
    logger.removeHandler(handler)

class TestSecretsInLogsExamplePublic(unittest.TestCase):
    def test_log_sensitive_public_runs(self):
        # Just ensure it runs as it is a demonstration
        log_sensitive_public()

if __name__ == "__main__":
    unittest.main()

File: RepoTransBench/target_projects/Python/Java/oxsecurity_MaskerLogger/src/test/java/com/oxsecurity/maskerlogger/publictests/TestPublicSecretsInLogsExample.java
package com.oxsecurity.maskerlogger.publictests;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;

import com.oxsecurity.maskerlogger.maskerformatter.MaskerFormatter;

import java.util.logging.*;

public class TestPublicSecretsInLogsExample {
    static void logSensitivePublic() {
        Logger logger = Logger.getLogger("test_logger_public");
        Handler handler = new StreamHandler(System.out, new MaskerFormatter("%(message)s", null));
        logger.addHandler(handler);
        logger.setLevel(Level.WARNING);
        logger.warning("secret field: thisshouldberedacted321 must be hidden");
        logger.warning("nondescript info log");
        // Simulate apply_mask = false: skip masking
        LogRecord record = new LogRecord(Level.WARNING, "skip me");
        record.setResourceBundleName("skip_mask");
        handler.publish(record);
        logger.removeHandler(handler);
    }

    @Test
    public void testLogSensitivePublicRuns() {
        logSensitivePublic();
    }
}